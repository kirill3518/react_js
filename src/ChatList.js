import { Link, Outlet } from "react-router-dom";

const chats = [
    {
        name: "Chat1",
        id: "chat1"
    },
    {
        name: "Chat2",
        id: "chat2"
    },
    {
        name: "Chat3",
        id: "chat3"
    }
];

export const ChatList = () => {
    return (
        <>
            <div className="chat-list">
                {chats.map((chat) => (
                    <Link to={`/chat/${chat.id}`} key={chat.id}>
                        <div>{chat.name}</div>
                    </Link>
                ))}
            </div >
            <Outlet />
        </>
    )
};