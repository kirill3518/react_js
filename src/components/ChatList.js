import '../App.styles.scss';
import { Link, Outlet } from "react-router-dom";
import InputForm from './InputForm';
import { useTheme, ThemeProvider } from '@mui/material/styles';

export const ChatList = ({ chats, addChat, deleteChat }) => {
    // console.log(`(1) chats = ${chats}`, `; (2) addChat = ${addChat}`, `; (3) deleteChat = ${deleteChat}`);
    console.log(`(1) chats.length = ${chats.length}`);
    console.log(chats);

    const handleSubmit = (newChatName) => {
        console.log(`newChatName=${newChatName}`);
        const newChat = {
            name: newChatName,
            // id: `chat-${Date.now()}`
            id: `chat${chats.length}`
        };
        addChat(newChat);
    };

    const theme = useTheme();

    return (
        <>
            <div className="App-header">
                <ThemeProvider theme={theme}>
                    {chats.map((chat) => (
                        <div key={chat.id}>
                            <Link to={`/chat/${chat.id}`}>
                                {chat.name}
                            </Link>
                            <span onClick={() => deleteChat(chat.id)}>delete</span>
                        </div>
                    ))}
                </ThemeProvider>
            </div >
            <div>Add Chat</div>
            <InputForm onAddMessage={handleSubmit} />
            <Outlet />
        </>
    )
};