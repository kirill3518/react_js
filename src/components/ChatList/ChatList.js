import './ChatList.styles.scss';
import { Link, Outlet } from "react-router-dom";
import InputForm from '../InputForm';

export const ChatList = ({ chats, handleSubmit, deleteChat }) => {

    return (
        <>
            <div className="App-header ChatList-header">
                <div>
                    <p>Add Chat</p>
                    {chats.map((chat) => (
                        <div key={chat.id}>
                            <Link to={`/chat/${chat.id}`}>
                                {chat.name}
                            </Link>
                            <span onClick={() => deleteChat(chat.id)} className="Delete-chat">delete</span>
                        </div>
                    ))}
                    <InputForm onAddMessage={handleSubmit} />
                </div>
                <Outlet />
            </div >
        </>
    )
};