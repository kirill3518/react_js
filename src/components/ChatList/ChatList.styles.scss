.ChatList-header {
    display: flex;

    .Delete-chat {
        margin-left: 10px;
    }
}