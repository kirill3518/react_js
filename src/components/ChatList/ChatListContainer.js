import { selectChats } from '../../store/chat/selectors';
import { shallowEqual, useSelector } from 'react-redux';
import { ChatList } from './ChatList';
import { useEffect, useState } from 'react';
import { onValue } from 'firebase/database';
import { chatsRef } from '../../services/firebase';

export const ChatListContainer = ({ addChat, deleteChat }) => {

    const [chats, setChats] = useState([]);
    // const chats = useSelector(selectChats, shallowEqual);

    console.log(chats);

    const handleSubmit = (newChatName) => {
        const newChat = {
            name: newChatName,
            id: `chat-${Date.now()}`
            // id: `chat${chats.length}`
        };
        addChat(newChat);
    };

    useEffect(() => {
        const unsubscribe = onValue(chatsRef, (snapshot) => {
            console.log(snapshot.val());
            setChats(Object.values(snapshot.val() || {}));
        });
        return unsubscribe;
    }, []);

    return (
        <>
            <ChatList chats={chats} handleSubmit={handleSubmit} deleteChat={deleteChat} />
        </>
    )
};