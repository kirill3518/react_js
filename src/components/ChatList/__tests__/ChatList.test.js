import { render } from "@testing-library/react";
import { ChatList } from "../ChatList";
import { ChatListContainer } from "../ChatListContainer";

describe("Компонент ChatList", () => {
    it("Рендер контейнера ChatListContainer", () => {
        const addNewChat = () => { };
        const removeChat = () => { };
        const component = render(
            <ChatListContainer addChat={addNewChat} deleteChat={removeChat} />
        );

        const text = component.getAllByText("Add Chat");
        expect(text).toBeDefined();
    });

    it("Снапшот презентационного ChatList", () => {
        const handleSubmit = () => { };
        const deleteChat = () => { };
        const chats = [];
        const component = render(
            <ChatList chats={chats} handleSubmit={handleSubmit} deleteChat={deleteChat} />
        );

        expect(component).toMatchSnapshot();
    });
});