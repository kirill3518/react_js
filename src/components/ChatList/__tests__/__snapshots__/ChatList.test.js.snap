// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент ChatList Снапшот презентационного ChatList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App-header ChatList-header"
      >
        <div>
          <p>
            Add Chat
          </p>
          <div
            class="InputForm"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              style="margin: 20px;"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Outlined
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-basic"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1in441m"
                  >
                    <span>
                      Outlined
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              style="margin: 20px;"
              tabindex="0"
              type="button"
            >
              Send
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App-header ChatList-header"
    >
      <div>
        <p>
          Add Chat
        </p>
        <div
          class="InputForm"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="margin: 20px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Outlined
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Outlined
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            style="margin: 20px;"
            tabindex="0"
            type="button"
          >
            Send
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
